### ISSUE TITLE
CI should test 16kb and non-16kb page alignment in libraries, packaging, and emulator

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#16917]
with upcoming need to be ["16kb alignment clean"](https://developer.android.com/guide/practices/page-sizes) the CI matrix for emulator tests should be updated to run against a 16kb emulator as well as a regular emulator. Suggest using this API/image-name tuple:

android-35;google_apis_ps16k

The page size should be verified in CI prior to execution, via the tests on linked google doc

Additionally, the .apk should be extracted and have .so file verification of page-alignment checked, and finally the zipalign should be checked - otherwise we won't know if we have actually done the work or not, and we won't track any random regressions

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android-Backend/pull/402#discussion_r1723732098_

**Comment by Rohanraj123** [comment]
I am working on this issue!!

**Comment by mikehardy** [comment]
Should be automated, but it's worth mentioning that I run it on both types of emulators from time to time and the app runs. So we are clean currently, we just don't test it in CI

**Comment by Haz3-jolt** [comment]
Can I work on this?

**Comment by david-allison** [comment]
@Haz3-jolt When you have a second, could you summarize how this is blocked, either here or on the linked PR

**Comment by Haz3-jolt** [comment]
We need to test on 16kb page size emulators, but the runners we use [android-emulator-runner](https://github.com/ReactiveCircus/android-emulator-runner), which doesn't yet have a release with the PR adding in support for 16kb emulators.

**Comment by mikehardy** [comment]
@david-allison @Haz3-jolt the suggested workaround was to depend directly on the commit in the upstream action from where they merged the PR I posted to them that implemented the support we need

I did that and the PR is running CI now, hopefully it finishes up... https://github.com/ankidroid/Anki-Android/pull/18248

no need to wait on upstream release when we've got a commit hash for them

**Comment by mikehardy** [comment]
Well, that's disappointing. `@actions/core not found` !? so CI still failed when it tried to run that action based on the commit hash, which is unexpected. So this is still blocked until they release upstream or we figure out how to make that action run from a commit hash

**Comment by Haz3-jolt** [comment]
> Well, that's disappointing. `@actions/core not found` !? so CI still failed when it tried to run that action based on the commit hash, which is unexpected. So this is still blocked until they release upstream or we figure out how to make that action run from a commit hash

I'm pretty sure this is because github actions really does not like it when we run an action on a commit hash. I've faced similar issues with a friend's project.

**Comment by mikehardy** [comment]
PR to do this would work if the upstream emulator-action would merge my PR. Pinged maintainer there
